### PR TITLE
Make client_mutation_id input field optional when defining schema for Relay Modern

### PR DESCRIPTION
**Pull request by hisapy**
Hi guys,

It's me again trying to contribute. This time I got the desired behavior making `clientMutationId` optional for Modern but keeping the required constraint for Classic.

This is just the beginning to support different Modern behaviors that can be added later without affecting the Classic stuff.

Examples:

```elixir
# Using in schema
use Absinthe.Relay.Schema, :modern

# Or if using Notation
use Absinthe.Relay.Schema.Notation, :modern
```

If you want to keep it **Classic** just don't add `:modern` when using it.

NOTICE: I need guidance on stuff to change to keep `@doc`s and `@moduledoc`s up-to-date.

PS.: I'm already using this in another project and works as expected.

Regards,

Hisa

### CODE REVIEW

File: lib/absinthe/relay/mutation/modern_notation.ex [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule Absinthe.Relay.Mutation.ModernNotation do
+  use Absinthe.Relay.Mutation.Base
+
+  defp client_mutation_id_field do
```
Comment (hisapy) [inline lib/absinthe/relay/mutation/modern_notation.ex]: This is how we override the default behavior.

**Comment by bruce** [comment]
Hey @hisapy, sorry it's taken so long to give this a look.

I think the feeling/direction behind this is good. I'm going to play around with the naming and structure a bit, and see if we can solidify the API on this a bit.

I think this serves as a good base for a `v1.4` release; the changes here could serve as a good foundation for targeted Modern vs Classic features in the future.

Also, I think we should probably output a deprecation message for the default `:classic` behavior now, warning that in `v1.5` it will switch to `:modern`.

**Comment by hisapy** [comment]
Hello @bruce, no problem at all ... It's you job as a maintainer to schedule things but let me know if you have any task you think I could help you with.

Perhaps for 1.5 you might want to _swap_ the behavior added in this pull request, i.e.: `use Absinthe.Schema.Notation :classic` to integrate with Relay Classic and thus, being `:modern` the default.

**Comment by bruce** [comment]
Going to merge this into our `next` branch and then work on some additional structural changes (just some module renaming and stuff) for inclusion in a v1.4.0-beta.0.

**Comment by tlvenn** [comment]
Great, would be awesome to make the input optional as well in Relay modern so that I can get rid of the fork I created.

**Comment by bruce** [comment]
@tlvenn Which fork/place should I look at if I was looking for how you need things for Modern?

**Comment by tlvenn** [comment]
@bruce I did the following https://github.com/jumpn/absinthe_relay/commit/a7f8da0f9146b6f37fe1338600f2e1bfc3eb694b to make the input optional

**Pull request merged**